halo2_proofs/CHANGELOG.md

```diff
@@ -6,54 +6,95 @@ and this project adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [0.1.0-beta.2] - 2022-02-14
```

Comment:
Some of the changes below are just readability refactors, but I also added extra changes that I extracted from the diff between [`halo2 0.1.0-beta.1` and current `main`](https://github.com/zcash/halo2/compare/3850b9ccd540fa38d7da4eec64ce7bd869898b42..247cd620eeb434e7f86dc8579774b30384ae02b2).